cmake/build_common.cmake

```diff
@@ -0,0 +1,108 @@
+if(NOT DEFINED ROOT OR NOT DEFINED ARCH)
+    message(FATAL_ERROR "Assert: ROOT = ${ROOT}; ARCH = ${ARCH}")
+endif()
+
+include("${CMAKE_CURRENT_LIST_DIR}/Version.cmake")
+
+
+set(BUILD 0)
+if(DEFINED ENV{BUILD_NUMBER})
+    set(BUILD $ENV{BUILD_NUMBER})
+endif()
+
+set(TAG "")
+if(DEFINED ENV{TAG})
+    set(TAG "$ENV{TAG}")
+endif()
+
+set(BUILD_PATH "${CMAKE_CURRENT_LIST_DIR}/../build")
+set(PACKAGE_NAME "cryptopp-${VERSION}-${ARCH}-${BUILD}${TAG}")
+set(CMAKE_INSTALL_PREFIX ${ROOT}/${PACKAGE_NAME})
+
+set(GNU_RELEASE_LIBS_NAME "libcryptopp")
+set(GNU_DEBUG_LIBS_NAME ${GNU_RELEASE_LIBS_NAME}d)
+
+set(MSVS_RELEASE_LIBS_NAME "cryptlib")
+set(MSVS_DEBUG_LIBS_NAME ${MSVS_RELEASE_LIBS_NAME}d)
+
+if(WIN32)
+    message(STATUS ${ARCH})
+    if(NOT ${ARCH} MATCHES x86)
+        set(ARCH x64)
+    endif()
+
+    execute_process(COMMAND msbuild cryptlib.vcxproj /p:Configuration=Release /p:Platform=${ARCH} #msbuild cryptlib.vcxproj /p:Configuration=Debug
+	WORKING_DIRECTORY "${BUILD_PATH}/.."
+    )
+
+    execute_process(COMMAND msbuild cryptlib.vcxproj /p:Configuration=Debug /p:Platform=${ARCH} #msbuild cryptlib.vcxproj /p:Configuration=Release
+	WORKING_DIRECTORY "${BUILD_PATH}/.."
+    )
+
+
+    execute_process(COMMAND ${CMAKE_COMMAND} -E copy ${ARCH}/Output/Release/${MSVS_RELEASE_LIBS_NAME}.lib ${ROOT}/${PACKAGE_NAME}/lib/${MSVS_RELEASE_LIBS_NAME}.lib
+	    COMMAND ${CMAKE_COMMAND} -E copy ${ARCH}/Output/Debug/${MSVS_RELEASE_LIBS_NAME}.lib ${ROOT}/${PACKAGE_NAME}/lib/${MSVS_DEBUG_LIBS_NAME}.lib
+	    COMMAND ${CMAKE_COMMAND} -E copy ${ARCH}/Output/Debug/${MSVS_RELEASE_LIBS_NAME}.pdb ${ROOT}/${PACKAGE_NAME}/lib/${MSVS_DEBUG_LIBS_NAME}.pdb
+	    WORKING_DIRECTORY "${BUILD_PATH}/.."
+    )
```

Comment:
why not use `file(COPY ...`?  
I suggest to remove these vars and use lib-file names explicitly.
Pdb for release has to be copied too

Reply:
Fixed. Thanks!